fuse_core/test/test_throttled_callback.cpp

```diff
@@ -227,35 +255,127 @@ TEST(ThrottledCallback, DropMessagesIfThrottlePeriodIsGreaterThanPublishPeriod)
+  timestamp.fromSec(kBaseTimeS - kEpsilonS);
+  unsigned i = 0;
+  for (; i < kPublishToThrottleRatio; ++i)
+  {
+    timestamp.fromSec(timestamp.toSec() + kPublishPeriodS);
+    publisher.publishWithHeaderTimestamp(timestamp);
+  }
+
+  ros::Duration(0.1).sleep();
+  EXPECT_EQ(sensor_model.getKeptMessages(), 0);
+  EXPECT_EQ(sensor_model.getDroppedMessages(), i);
+
+  // Publish for another period, phase aligned to just before the period division
+  // Should keep one message
+  sensor_model.reset();
+
+  for (; i < 2 * kPublishToThrottleRatio; ++i)
+  {
+    timestamp.fromSec(timestamp.toSec() + kPublishPeriodS);
+    publisher.publishWithHeaderTimestamp(timestamp);
+  }
+  ros::Duration(0.1).sleep();
+
+  EXPECT_EQ(sensor_model.getKeptMessages(), 1);
+  EXPECT_EQ(sensor_model.getDroppedMessages(), kPublishToThrottleRatio - 1);
+
+  // Publish one more message which puts us over the next period division, should be kept
+  sensor_model.reset();
+  timestamp.fromSec(timestamp.toSec() + kPublishPeriodS);
+  publisher.publishWithHeaderTimestamp(timestamp);
+  ros::Duration(0.1).sleep();
+
+  EXPECT_EQ(sensor_model.getKeptMessages(), 1);
+  EXPECT_EQ(sensor_model.getDroppedMessages(), 0);
+}
+
+// Test that the call operator is behaving correctly
+TEST(ThrottledCallback, TestCallOperator)
+{
+  constexpr double kThrottlePeriod = 0.1;
+
+  unsigned base_kept = 0;
+  auto base_keep_callback = [&base_kept](const geometry_msgs::PointStamped::ConstPtr & msg){ ++base_kept; };
+
+  unsigned header_timestamp_kept = 0;
+  auto msg_keep_callback =
+    [&header_timestamp_kept](const geometry_msgs::PointStamped::ConstPtr & msg){ ++header_timestamp_kept; };
+
+  fuse_core::ThrottledCallback<std::function<void(const typename geometry_msgs::PointStamped::ConstPtr&)>>
+    base_callback(base_keep_callback, nullptr, ros::Duration(kThrottlePeriod));
+
+  fuse_core::ThrottledMessageCallback<geometry_msgs::PointStamped> message_header_calback(
+    msg_keep_callback, nullptr, ros::Duration(kThrottlePeriod), false, true);
+
+  geometry_msgs::PointStamped point_msg;
+  point_msg.header.stamp.fromSec(0.0);
+
+  geometry_msgs::PointStampedConstPtr msg_const = boost::make_shared<const geometry_msgs::PointStamped>(point_msg);
+
+  // Initialize both callbacks
+  base_callback(msg_const);
+  message_header_calback(msg_const);
+
+  // Neither keeps the first message
+  EXPECT_EQ(base_kept, 0);
+  EXPECT_EQ(header_timestamp_kept, 0);
+
+  ros::Duration(1.0).sleep();
+  // Call with the same message again
+  base_callback(msg_const);
+  message_header_calback(msg_const);
+  // The base callback should keep this one because it's related to elapsed time
+  EXPECT_EQ(base_kept, 1);
+  // The header callback should not keep this one because the header timestamp is the same as the old one
+  EXPECT_EQ(header_timestamp_kept, 0);
+
+  // Advance the header timestamp and check that the header timestamp based throttle keeps it
+  point_msg.header.stamp += ros::Duration().fromSec(1.0);
```

Comment:
No need to call `fromSec`
```suggestion
  point_msg.header.stamp += ros::Duration(1.0);
```